beacon-chain/sync/data_columns_sampling.go

```diff
@@ -4,24 +4,222 @@ import (
 	"context"
 	"fmt"
 	"sort"
+	"sync"
+	"time"
 
 	"github.com/btcsuite/btcd/btcec/v2"
 	"github.com/ethereum/go-ethereum/common/math"
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/libp2p/go-libp2p/core/peer"
 	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/prysmaticlabs/prysm/v5/async"
 	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/feed"
 	statefeed "github.com/prysmaticlabs/prysm/v5/beacon-chain/core/feed/state"
 	"github.com/prysmaticlabs/prysm/v5/beacon-chain/core/peerdas"
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/p2p"
 	"github.com/prysmaticlabs/prysm/v5/beacon-chain/p2p/types"
 	fieldparams "github.com/prysmaticlabs/prysm/v5/config/fieldparams"
 	"github.com/prysmaticlabs/prysm/v5/config/params"
 	"github.com/prysmaticlabs/prysm/v5/crypto/rand"
 	eth "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
 	"github.com/prysmaticlabs/prysm/v5/runtime/version"
-	"github.com/sirupsen/logrus"
 )
 
+const PeerRefreshInterval = 1 * time.Minute
+
+// DataColumnSampler defines the interface for sampling data columns from peers for requested block root and samples count.
+type DataColumnSampler interface {
+	// Run starts the data column sampling service.
+	Run(ctx context.Context, stateNotifier statefeed.Notifier)
+}
+
+var _ DataColumnSampler = (*DataColumnSampler1D)(nil)
+
+// DataColumnSampler1D is a 1D data column sampler for PeerDAS 1D.
+type DataColumnSampler1D struct {
+	sync.RWMutex
```

Comment:
You'll need to use this lock to protect access to `peerToColumnMap` and `columnToPeerMap`.